### ISSUE TITLE
Error on save() when observing hasManyFragments property

### ISSUE THREAD

**Issue by dwickern** [lytics/ember-data.model-fragments#48]
Live example (hit save): http://jsbin.com/yibuniqami/1/edit?js,console,output

This happens when observing a hasManyFragments array, e.g.

`whoops: function() {}.observes('hasManyFragmentProperty.[]')`

My REST API responds with the server's version of the model. The client chokes with this error when it tries to update its version
```
TypeError: Cannot read property 'destroy' of undefined
    at ContainerView.extend.arrayWillChange (http://builds.emberjs.com/tags/v1.9.0/ember.js:40448:20)
    at applyStr (http://builds.emberjs.com/tags/v1.9.0/ember.js:19670:29)
    at sendEvent (http://builds.emberjs.com/tags/v1.9.0/ember.js:14106:13)
    at __exports__.default.Mixin.create.arrayContentWillChange (http://builds.emberjs.com/tags/v1.9.0/ember.js:30434:9)
    at EmberObject.extend.arrangedContentArrayWillChange (http://builds.emberjs.com/tags/v1.9.0/ember.js:34301:14)
    at applyStr (http://builds.emberjs.com/tags/v1.9.0/ember.js:19670:29)
    at sendEvent (http://builds.emberjs.com/tags/v1.9.0/ember.js:14106:13)
    at Array.__exports__.default.Mixin.create.arrayContentWillChange (http://builds.emberjs.com/tags/v1.9.0/ember.js:30434:9)
    at Array.Mixin.create.replace (http://builds.emberjs.com/tags/v1.9.0/ember.js:35815:14)
    at StatefulArray.extend.replaceContent (https://rawgit.com/lytics/ember-data.model-fragments/master/dist/ember-data.model-fragments.js:312:37)
```

If I change from observing `[]` to `@each` I get this error instead:
```
RangeError: Maximum call stack size exceeded
    at GETTER_FUNCTION [as _parentView] (http://builds.emberjs.com/tags/v1.9.0/ember.js:17067:38)
    at null.<anonymous> (http://builds.emberjs.com/tags/v1.9.0/ember.js:41363:26)
    at Descriptor.ComputedPropertyPrototype.get (http://builds.emberjs.com/tags/v1.9.0/ember.js:12122:25)
    at get (http://builds.emberjs.com/tags/v1.9.0/ember.js:17570:21)
    at CoreView.extend.destroy (http://builds.emberjs.com/tags/v1.9.0/ember.js:43560:36)
    at apply (http://builds.emberjs.com/tags/v1.9.0/ember.js:19652:32)
    at superWrapper [as destroy] (http://builds.emberjs.com/tags/v1.9.0/ember.js:19226:15)
    at ContainerView.extend.arrayWillChange (http://builds.emberjs.com/tags/v1.9.0/ember.js:40448:21)
    at applyStr (http://builds.emberjs.com/tags/v1.9.0/ember.js:19670:29)
    at sendEvent (http://builds.emberjs.com/tags/v1.9.0/ember.js:14106:13)
```

I can work around it by removing the fragments from the response in the adapter.

**Comment by slindberg** [comment]
Is happening after a recent update, or are you using model fragments for the first time?

**Comment by slindberg** [comment]
Nevermind, I see that it only happens with ED beta.14 + EDMF 0.2.5. This will likely take a while to track down. If you are able to create a failing test, you'd be my hero :metal:

**Comment by slindberg** [comment]
Fixed by #54
